### DISCUSSION TITLE
How to avoid corruption when working with database schema managed externally

### DISCUSSION THREAD

**Discussion by lachance-martin** [directus/directus#8901] [Archive]
I am trying out Directus and it seems almost perfect for our needs. Great product! The only thing I could not figure out yet is how to manage database migrations.

We have a database that continually evolves (with Liquibase scripts). So obviously it is managed outside of Directus. When removing/renaming columns/tables, it seems that it corrupts Directus.

Is there a way to tell Directus to reinspect the new database schema, and drop what is now irrelevant?

**Comment by rijkvanzanten** [comment]
There's no thing in place to auto-cleanup any system meta rows after the raw db structure has changed. We're discussing a command for that over here: https://github.com/directus/directus/discussions/8652. For this particular use case, it sounds like it's most useful to auto-drop any meta rows that are referring a non-existent table/column/relation
